### PR TITLE
[FLINK-28793] Allow to GetInfo for HiveServer2 Endpoint

### PR DESCRIPTION
**Pull request by chaozwn**
## What is the purpose of the change

allow to GetInfo for HiveServer2 Endpoint.

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

This change is already covered by existing tests,.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-sql-gateway-api/src/main/java/org/apache/flink/table/gateway/api/SqlGatewayService.java [hunk 1]
```diff
@@ -187,6 +187,13 @@ ResultSet fetchResults(
             int maxRows)
             throws SqlGatewayException;
 
+    /**
+     * Describe the cluster info.
+     *
+     * @return Returns gateway info.
+     */
+    Map<String, String> getGatewayInfo();
```
Comment (fsk119) [inline flink-table/flink-sql-gateway-api/src/main/java/org/apache/flink/table/gateway/api/SqlGatewayService.java]: I think it's better to use POJO to describe all infos.

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java [hunk 2]
```diff
@@ -500,19 +504,43 @@ private void runGetObjectTest(
         }
     }
 
+    @Test
+    public void testGetInfo() throws Exception {
```
Comment (fsk119) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/table/endpoint/hive/HiveServer2EndpointITCase.java]: it's better to use jdbc as the test utils.

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. I add a commit to resolve my concern.

**Comment by chaozwn** [comment]
LGTM.

**Review by fsk119 — APPROVED** [review]
Thanks for the feedback. LGTM.

Merging...

**Pull request closed without merging**